google/resource_logging_project_sink.go

```diff
@@ -0,0 +1,147 @@
+package google
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/logging/v2"
+)
+
+const nonUniqueWriterAccount = "serviceAccount:cloud-logs@system.gserviceaccount.com"
+
+func resourceLoggingProjectSink() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLoggingProjectSinkCreate,
+		Read:   resourceLoggingProjectSinkRead,
+		Delete: resourceLoggingProjectSinkDelete,
+		Update: resourceLoggingProjectSinkUpdate,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"destination": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"filter": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"unique_writer_identity": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+
+			"writer_identity": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceLoggingProjectSinkCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	name := d.Get("name").(string)
+
+	id := LoggingSinkId{
+		typ:     "projects",
+		typName: project,
+		name:    name,
+	}
+
+	sink := logging.LogSink{
+		Name:        d.Get("name").(string),
+		Destination: d.Get("destination").(string),
+	}
+	if v, ok := d.GetOk("filter"); ok {
```

Comment:
Do you need this if block? d.Get("filter") will return `""` and should work without this extra if block

Reply:
That's easier to read. Fixed.